processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -578,6 +579,43 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
```

Comment:
Actually… we don't need `dimensionOrder` at all for this, we can just return `dimensions` itself. So this can return a `List<String>` that is actually just `ImmutableList.copyOf(dimensions)`.

That also makes the loading simpler since you can pass that list directly to the one that takes an Iterable.